diff --git a/sasl/saslwrapper.h b/sasl/saslwrapper.h
index 099ed22..bb94d5e 100644
--- a/sasl/saslwrapper.h
+++ b/sasl/saslwrapper.h
@@ -20,6 +20,7 @@
 #include <stdint.h>
 #include <string>
 #include <sasl/sasl.h>
+#include <sasl/saslutil.h>
 #include <sstream>
 #include <stdlib.h>
 #include <string.h>
diff --git a/setup.py b/setup.py
index 85e40c9..20728d4 100644
--- a/setup.py
+++ b/setup.py
@@ -37,12 +37,18 @@ if sys.platform == 'darwin':
         if python_target < '10.9' and current_system >= '10.9':
             os.environ['MACOSX_DEPLOYMENT_TARGET'] = '10.9'
 
+macros = []
+libs = ["sasl2"]
+if sys.platform.startswith("win"):
+    macros = [('WIN32', '1')]
+    libs = ["libsasl"]
 
 sasl_module = Extension('sasl.saslwrapper',
                         sources=['sasl/saslwrapper.cpp'],
                         include_dirs=["sasl"],
-                        libraries=["sasl2"],
-                        language="c++")
+                        libraries=libs,
+                        language="c++",
+                        define_macros=macros)
 setup(name='sasl',
       version='0.3.1',
       url="http://github.com/cloudera/python-sasl",
