Index: work/sasl/saslwrapper.cpp
===================================================================
--- work.orig/sasl/saslwrapper.cpp
+++ work/sasl/saslwrapper.cpp
@@ -2766,7 +2766,7 @@ static PyTypeObject __pyx_type_4sasl_11s
   sizeof(struct __pyx_obj_4sasl_11saslwrapper_Client), /*tp_basicsize*/
   0, /*tp_itemsize*/
   __pyx_tp_dealloc_4sasl_11saslwrapper_Client, /*tp_dealloc*/
-  0, /*tp_print*/
+  0, /*tp_print or tp_vectorcall_offset*/
   0, /*tp_getattr*/
   0, /*tp_setattr*/
   #if PY_MAJOR_VERSION < 3
@@ -2998,7 +2998,9 @@ PyMODINIT_FUNC PyInit_saslwrapper(void)
   __pyx_vtable_4sasl_11saslwrapper_Client.getSSF = (PyObject *(*)(struct __pyx_obj_4sasl_11saslwrapper_Client *, int __pyx_skip_dispatch))__pyx_f_4sasl_11saslwrapper_6Client_getSSF;
   __pyx_vtable_4sasl_11saslwrapper_Client.getError = (PyObject *(*)(struct __pyx_obj_4sasl_11saslwrapper_Client *, int __pyx_skip_dispatch))__pyx_f_4sasl_11saslwrapper_6Client_getError;
   if (PyType_Ready(&__pyx_type_4sasl_11saslwrapper_Client) < 0) {__pyx_filename = __pyx_f[0]; __pyx_lineno = 48; __pyx_clineno = __LINE__; goto __pyx_L1_error;}
+#if PY_VERSION_HEX < 0x030800b4
   __pyx_type_4sasl_11saslwrapper_Client.tp_print = 0;
+#endif
   if (__Pyx_SetVtable(__pyx_type_4sasl_11saslwrapper_Client.tp_dict, __pyx_vtabptr_4sasl_11saslwrapper_Client) < 0) {__pyx_filename = __pyx_f[0]; __pyx_lineno = 48; __pyx_clineno = __LINE__; goto __pyx_L1_error;}
   if (PyObject_SetAttrString(__pyx_m, "Client", (PyObject *)&__pyx_type_4sasl_11saslwrapper_Client) < 0) {__pyx_filename = __pyx_f[0]; __pyx_lineno = 48; __pyx_clineno = __LINE__; goto __pyx_L1_error;}
   __pyx_ptype_4sasl_11saslwrapper_Client = &__pyx_type_4sasl_11saslwrapper_Client;
